unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FeatureController/FeatureController.cs

```diff
@@ -32,27 +40,60 @@ public void Update()
         }
     }
 
+    public void LateUpdate()
+    {
+        foreach (Feature feature in activeFeatures)
+        {
+            feature.LateUpdate();
+        }
+    }
+
+    public void OnDestroy()
+    {
+        foreach (Feature feature in activeFeatures)
+        {
+            feature.Dispose();
+        }
+    }
+
     public void OnKernelConfigChanged(KernelConfigModel current, KernelConfigModel previous) { ApplyFeaturesConfig(current); }
 
     public void ApplyFeaturesConfig(KernelConfigModel config)
     {
-        HandleBuilderInWorld(config.features.enableBuilderInWorld);
+        HandleFeature<BIWMainController>(config.features.enableBuilderInWorld);
         currentConfig = config;
     }
 
-    private void HandleBuilderInWorld(bool isActive)
+    private void HandleFeature<T>(bool isActive) where T : Feature
     {
         if (isActive)
+            InitializeFeature<T>();
+        else
+            RemoveFeature<T>();
+    }
+
+    private void InitializeFeature<T>() where T : Feature
+    {
+        for (int i = 0; i < activeFeatures.Count; i++)
         {
-            if (builderInWorld != null)
+            if (activeFeatures[i].GetType() == typeof(T))
                 return;
-            builderInWorld = GameObject.Instantiate(builderInWorldFeaturePrefab);
-
         }
-        else
+
+        Feature feature = (Feature) Activator.CreateInstance(typeof (T));
```

Comment:
Be aware that `Activator.CreateInstance` is excruciatingly slow. We can avoid it by using the `new()` generic constraint.

```
private void InitializeFeature<T>() where T : Feature, new ()
{
    ...
    
    Feature feature = (Feature) new T();
}
```

In any case we should revise this code on the future, because we'd want to use custom constructors to DI. So here we most likely need to use an abstract factory, or just pass the new `Feature` object by argument.

Reply:
Changed and added a note in the code to have this into account. Should I create a debt tech tiquet?